agent/config/config.go

```diff
@@ -312,6 +217,147 @@ func get(args []string, l *logrus.Entry) (cfg *Config, configFileF string, err e
 	return //nolint:nakedret
 }
 
+func setDefaults(cfg *Config, l *logrus.Entry) {
+	if cfg == nil {
+		return
+	}
+
+	// set default values
+	configureNetworkDefaults(cfg)
+
+	if cfg.WindowConnectedTime == 0 {
+		cfg.WindowConnectedTime = time.Hour
+	}
+
+	configurePaths(cfg, l)
+
+	if cfg.Server.Address != "" {
+		if _, _, e := net.SplitHostPort(cfg.Server.Address); e != nil {
+			host := cfg.Server.Address
+			cfg.Server.Address = net.JoinHostPort(host, "443")
+			l.Infof("Updating PMM Server address from %q to %q.", host, cfg.Server.Address)
+		}
+	}
+
+	// enabled cross-component PMM_DEBUG and PMM_TRACE take priority
+	if b, _ := strconv.ParseBool(os.Getenv("PMM_DEBUG")); b {
+		cfg.Debug = true
+	}
+	if b, _ := strconv.ParseBool(os.Getenv("PMM_TRACE")); b {
+		cfg.Trace = true
+	}
+}
+
+var (
+	defaultListenSocketJSON         = "/usr/local/percona/pmm2/pmm-agent.sock"
+	defaultListenSocketGRPCFilename = "pmm-agent-grpc.sock"
+	// DefaultListenSocketGRPC defines path to a socket on which GRPC server is listening.
+	DefaultListenSocketGRPC = "/usr/local/percona/pmm2/" + defaultListenSocketGRPCFilename
```

Comment:
we have `cfg.Paths.PathsBase`, let's put sockets in the same directory or into subdirectory of `cfg.Paths.PathsBase` like we do with `exporters`

Reply:
Wasn't aware of this. Great catch.